### PR TITLE
refactor Address, DomesticAddress, migrations

### PR DESCRIPTION
**Pull request by djamesfar**
- Changed addressable table to addresses (polymorphic)
- changed addresses to domestic_addresses
- updated permission migration
- updated models

### CODE REVIEW

File: database/migrations/2015_01_01_130000_create_zips_table.php [hunk 1]
```diff
@@ -16,8 +16,8 @@ public function up()
             $table->foreignIdFor(State::class);
             $table->foreignIdFor(Region::class)->nullable();
             $table->foreignIdFor(Timezone::class)->nullable();
-            $table->decimal('latitude', 4, 2)->nullable();
-            $table->decimal('longitude', 5, 2)->nullable();
+            $table->decimal('latitude', 9, 6)->nullable();
+            $table->decimal('longitude', 8, 6)->nullable();
```
Comment (drewroberts) [inline database/migrations/2015_01_01_130000_create_zips_table.php:20]: I don't know if a zip code can have that specific of a longitude & latitude but I will see what I can get in the database

File: src/Models/Address.php [hunk 2]
```diff
@@ -5,38 +5,29 @@
...
 
 use Tipoff\Support\Models\BaseModel;
+use Tipoff\Support\Traits\HasCreator;
 use Tipoff\Support\Traits\HasPackageFactory;
+use Tipoff\Support\Traits\HasUpdater;
 
 class Address extends BaseModel
 {
+    use HasCreator;
     use HasPackageFactory;
```
Comment (drewroberts) [inline src/Models/Address.php:15]: The Address model doesn't have a factory yet. Can you add it to this PR, @djamesfar?

File: src/AddressesServiceProvider.php [hunk 3]
```diff
@@ -27,7 +27,7 @@ public function configureTipoffPackage(TipoffPackage $package): void
     {
         $package
             ->hasPolicies([
-                Address::class => AddressPolicy::class,
+                DomesticAddress::class => AddressPolicy::class,
```
[the remark this answers is not in the document]
Comment (djamesfar) [reply src/AddressesServiceProvider.php:30]: :+1:

[the remark this answers is not in the document]
Comment (djamesfar) [reply src/AddressesServiceProvider.php:30]: I was thinking the permissions would be the exact same for all addresses?

**Comment by drewroberts** [comment]
@djamesfar We also need a nova resource for Address

File: src/Models/Address.php (hunk 2 above)
Comment (drewroberts) [reply src/Models/Address.php:15]: Thank you

File: src/AddressesServiceProvider.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (drewroberts) [reply src/AddressesServiceProvider.php]: There will likely be cases where an application doesn't allow a Domestic Address to be updated but will allow Staff to update an Address.

**Review by drewroberts — APPROVED** [review]
Thank you! 🍭

**Pull request merged**